cryptoki/src/mechanism/hkdf.rs

```diff
@@ -0,0 +1,116 @@
+//! Mechanisms of hash-based key derive function (HKDF)
```

Comment:
It'd be good to insert a copyright header here, just like so: https://github.com/parallaxsecond/rust-cryptoki/blob/main/cryptoki/src/types.rs#L1-L2

Reply:
@wiktor-k do we use to put 2021 everywhere or the year the file is added instead?

Reply:
Let's use 2024 here instead there, until we find a  better solution :)